src/js/auth.js

```diff
@@ -161,7 +161,11 @@ helper.hasPermission = (
   allowedPolicyNames,
   requestedOrgId,
 ) => {
-  if (userOrganization && requestedOrgId !== userOrganization.id) {
+  if (
+    userOrganization &&
+    requestedOrgId &&
```

Comment:
A valid `userOrganization` and a falsy `requestedOrgId` means that the user is trying to request data for _all_ orgs, not just their own.
This operation should be blocked.

Reply:
The requestedOrgId doesn't exist for the admin role.  Should we refactor admin login so we can require a requestedOrgId in all cases?

Reply:
Currently, all org requests for trees/planters follow the `/api/organization/{requestedOrgId}/trees|planter/` pattern, while all non-org requests use `/api/trees|planter/` (with no `requestedOrgId`).

The intention is that non-org users could make requests for specific orgs using the `/api/organization/{requestedOrgId}/...` pattern (in fact, this might be the better way of doing filtering by org in the Verify, Planters and Trees pages).

The best fix would be to implement the same organization-specific pattern for organization queries (possibly with a name change to reduce confusion).
There will soon be a new `stakeholder` endpoint in the Treetracker API (or possibly a separate service) that will allow us to query and manage organizations / entities / stakeholders, so I'm reluctant to do too much more on this in the legacy API.

What do you think?

Reply:
I misunderstood the intent as allowing admin to have access when the requestedOrgId was not provided.

Reply:
I tried out adding a new endpoint for organizations that would filter by entity.  I also updated the request from the client to pass along the orgId from the get-go.  It seems to be working so far.  I'll push the client changes in a little bit.